echopype/convert/api.py

```diff
@@ -0,0 +1,553 @@
+            if compress
+            else None,
+        )
+
+
+def _set_convert_params(param_dict):
+    """Set parameters (metadata) that may not exist in the raw files.
+
+    The default set of parameters include:
+    - Platform group: ``platform_name``, ``platform_type``, ``platform_code_ICES``, ``water_level``
+    - Platform/NMEA: ``nmea_gps_sentence``,
+                    for selecting specific NMEA sentences, with default values ['GGA', 'GLL', 'RMC'].
+    - Top-level group: ``survey_name``
+
+    Other parameters will be saved to the top level.
+
+    # TODO: revise docstring, give examples.
+    Examples
+    --------
+    # set parameters that may not already be in source files
+    echodata.set_param({
+        'platform_name': 'OOI',
+        'platform_type': 'mooring'
+    })
+    """
+    # TODO: revise docstring, give examples.
+    # TODO: need to check and return valid/invalid params as done for Process
+    out_params = dict()
+
+    # Parameters for the Platform group
+    out_params["platform_name"] = param_dict.get("platform_name", "")
+    out_params["platform_code_ICES"] = param_dict.get("platform_code_ICES", "")
+    out_params["platform_type"] = param_dict.get("platform_type", "")
+    out_params["water_level"] = param_dict.get("water_level", None)
+    out_params["nmea_gps_sentence"] = param_dict.get(
+        "nmea_gps_sentence", NMEA_SENTENCE_DEFAULT
+    )
+
+    # Parameters for the Top-level group
+    out_params["survey_name"] = param_dict.get("survey_name", "")
+    for k, v in param_dict.items():
+        if k not in out_params:
+            out_params[k] = v
+
+    return out_params
+
+
+def _check_file(file, model, xml_path=None, storage_options={}):
+
+    if MODELS[model]["xml"]:  # if this sonar model expects an XML file
+        if not xml_path:
+            raise ValueError(f"XML file is required for {model} raw data")
+        elif ".XML" not in os.path.splitext(xml_path)[1].upper():
+            raise ValueError(
+                f"{os.path.basename(xml_path)} is not an XML file"
+            )
+
+        xmlmap = fsspec.get_mapper(xml_path, **storage_options)
+        if not xmlmap.fs.exists(xmlmap.root):
+            raise FileNotFoundError(
+                f"There is no file named {os.path.basename(xml_path)}"
+            )
+
+        xml = xml_path
+    else:
+        xml = ""
+
+    # TODO: https://github.com/OSOceanAcoustics/echopype/issues/229
+    #  to add compatibility for pathlib.Path objects for local paths
+    fsmap = fsspec.get_mapper(file, **storage_options)
+    ext = MODELS[model]["ext"]
+    if not fsmap.fs.exists(fsmap.root):
+        raise FileNotFoundError(
+            f"There is no file named {os.path.basename(file)}"
+        )
+
+    if os.path.splitext(file)[1] != ext:
+        raise ValueError(
+            f"Not all files are in the same format. Expecting a {ext} file but got {file}"
+        )
+
+    return file, xml
+
+
+def open_raw(
+    file=None,
+    model=None,
```

Comment:
What do you think if we make `file` --> `raw_file` and `model` --> `sonar_model` ?  Seems better prompts for users, though it does mean changes in the tests.

Reply:
That's totally fine. I think it makes more sense that way. I will still have to fix and improve the tests, so don't worry about that :smile: